pkg/plugins/manager/client/client.go

```diff
@@ -101,35 +88,29 @@ func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceReq
 		return plugins.ErrPluginNotRegistered
 	}
 
-	totalBytes := float64(len(req.Body))
-	err := instrumentation.InstrumentCallResourceRequest(ctx, &req.PluginContext, instrumentation.Cfg{
-		Target: p.Target(),
-	}, totalBytes, func(ctx context.Context) (innerErr error) {
-		removeConnectionHeaders(req.Headers)
-		removeHopByHopHeaders(req.Headers)
-		removeNonAllowedHeaders(req.Headers)
-
-		processedStreams := 0
-		wrappedSender := callResourceResponseSenderFunc(func(res *backend.CallResourceResponse) error {
-			// Expected that headers and status are only part of first stream
-			if processedStreams == 0 && res != nil {
-				if len(res.Headers) > 0 {
-					removeConnectionHeaders(res.Headers)
-					removeHopByHopHeaders(res.Headers)
-					removeNonAllowedHeaders(res.Headers)
-				}
-
-				ensureContentTypeHeader(res)
+	// TODO: implement as middleware?
+	removeConnectionHeaders(req.Headers)
+	removeHopByHopHeaders(req.Headers)
+	removeNonAllowedHeaders(req.Headers)
+
+	processedStreams := 0
+	wrappedSender := callResourceResponseSenderFunc(func(res *backend.CallResourceResponse) error {
+		// Expected that headers and status are only part of first stream
+		if processedStreams == 0 && res != nil {
+			if len(res.Headers) > 0 {
+				removeConnectionHeaders(res.Headers)
+				removeHopByHopHeaders(res.Headers)
+				removeNonAllowedHeaders(res.Headers)
 			}
 
-			processedStreams++
-			return sender.Send(res)
-		})
+			ensureContentTypeHeader(res)
+		}
 
-		innerErr = p.CallResource(ctx, req, wrappedSender)
-		return
+		processedStreams++
+		return sender.Send(res)
 	})
```

Comment:
This should probably be its own middleware as well, but it can be left as a follow-up PR

Reply:
Not sure it should since this is kind of core functionality for the client to work as expected. We  might have a few similar middlewares, but I think we can keep this here at least for now.